ctapipe/io/dl1eventsource.py

```diff
@@ -0,0 +1,424 @@
+)
+
+
+class DL1EventSource(EventSource):
+    def __init__(
+        self,
+        input_url,
+        config=None,
+        parent=None,
+        **kwargs
+    ):
+        """
+        EventSource for dl1 files in the standard DL1 data format
+
+        Parameters:
+        -----------
+        input_url : str
+            Path of the file to load
+        config : traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            Set to None if no configuration to pass.
+        parent: ??
+            Parent from which the config is used. Mutually exclusive with config
+        kwargs
+        """
+        super().__init__(
+            input_url=input_url,
+            config=config,
+            parent=parent,
+            **kwargs
+        )
+
+        self.file_ = tables.open_file(input_url)
+        self.input_url = input_url
+        self._subarray_info = self._prepare_subarray_info()
+        self._mc_headers = self._parse_mc_headers()
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close()
+
+    def close(self):
+        self.file_.close()
+
+    @staticmethod
+    def is_compatible(file_path):
+        with open(file_path, 'rb') as f:
+            magic_number = f.read(8)
+        if magic_number != b'\x89HDF\r\n\x1a\n':
+            return False
+        with tables.open_file(file_path) as f:
+            metadata = f.root._v_attrs
+            if 'CTA PRODUCT DESCRIPTION' not in metadata._v_attrnames:
+                return False
+            if metadata['CTA PRODUCT DESCRIPTION'] != 'DL1 Data Product':
+                return False
+        return True
+
+    @property
+    def is_simulation(self):
+        """
+        True for files with a simulation group at the root of the file.
+        """
+        return "simulation" in self.file_.root
+
+    @property
+    def has_simulated_dl1(self):
+        """
+        True for files with telescope-wise event information in the simulation group
+        """
+        if self.is_simulation:
+            if "telescope" in self.file_.root.simulation.event:
+                return True
+        return False
+
+    @property
+    def subarray(self):
+        return self._subarray_info
+
+    @property
+    def datalevels(self):
+        params = 'parameters' in self.file_.root.dl1.event.telescope
+        images = 'images' in self.file_.root.dl1.event.telescope
+        if params and images:
+            return (DataLevel.DL1_IMAGES, DataLevel.DL1_PARAMETERS)
+        elif params:
+            return (DataLevel.DL1_PARAMETERS,)
+        elif images:
+            return (DataLevel.DL1_IMAGES,)
+
+    @property
+    def obs_id(self):
+        return set(self.file_.root.dl1.event.subarray.trigger.col("obs_id"))
+
+    @property
+    def mc_headers(self):
+        return self._mc_headers
+
+    @property
+    def obs_ids(self):
+        return self.mc_headers.keys()
```

Comment:
`obs_ids` should also work for observed data, this is not mc specific

Reply:
This might actually be an artifact. I dont see how this would be used over the obs_id property